lib/SPIRV/SPIRVReader.cpp

```diff
@@ -739,6 +739,14 @@ SPIRVToLLVM::getMetadataFromNameAndParameter(std::string Name,
               ConstantInt::get(Type::getInt32Ty(*Context), Parameter))};
 }
 
+inline std::vector<llvm::Metadata *>
+SPIRVToLLVM::getMetadataFromNameAndParameter(std::string Name,
+                                             int64_t Parameter) {
```

Comment:
Would it make sense to remove the preceding overload for `SPIRVWord` (a.k.a. `unsigned`) and only leave this `int64_t` signature? It's doubtable that we'd run into any unsigned token number that would exceed 32 bits...

Reply:
I wouldn't recommend to remove `SPIRVWord` overload, because `SPIRVWord` is directly mapped to the spec and widely used in the translator implementation. Moreover, mapping some 32-bit value down to 64-bit value might confuse SPIR-V -> LLVM IR -> SPIR-V roundtrip, because LLVM IR -> SPIR-V might expect 32-bit value.

Instead, I would suggest to accept two SPIR-V words here and automatically bundle them into a single 64-bit value within the function. Note: if we go this way, we should probably leave a comment near this function that it produces a metadata with a single value and not with two values

Reply:
During an offline discussion I was pointed out that this function is not always called and in order to determine whether it should be called or not, concatenation of two SPIR-V words should occur first.

Therefore, my second part of the comment should be ignored:

> Instead, I would suggest to accept two SPIR-V words here and automatically bundle them into a single 64-bit value within the function. Note: if we go this way, we should probably leave a comment near this function that it produces a metadata with a single value and not with two values

The first one is still stands